google/gax/api_callable.py

```diff
@@ -417,7 +439,9 @@ def base_caller(api_call, _, *args):
 
     def inner(request, options=None):
         """Invoke with the actual settings."""
-        this_settings = settings.merge(options)
+        this_options = _merge_options_metadata(options, settings)
```

Comment:
It's confusing that we now do two merges of settings/options: one for metadata, one for everything else. Can we do all of this in `settings.merge`?